pkg/jsonSchemaValidator/validator_test.go

```diff
@@ -0,0 +1,57 @@
+package jsonSchemaValidator
+
+import (
+	"testing"
+
+	"github.com/datreeio/datree/pkg/fileReader"
+	"github.com/ghodss/yaml"
+	"github.com/stretchr/testify/assert"
+)
+
+const yamlFilesPath = "../../internal/fixtures/policyAsCode/custom-keys"
+
+func TestValidateCustomKeysFail(t *testing.T) {
+	fileReader := fileReader.CreateFileReader(nil)
+
+	failResourceYamlFileContent, err := fileReader.ReadFileContent(yamlFilesPath + "/fail-yaml-file.yaml")
+	if err != nil {
+		panic(err)
+	}
+
+	customRuleSchemaYamlFileContent, err := fileReader.ReadFileContent(yamlFilesPath + "/custom-rule.yaml")
+	if err != nil {
+		panic(err)
+	}
+
+	jsonSchemaValidator := New()
+
+	resourceYaml, _ := yaml.JSONToYAML([]byte(failResourceYamlFileContent))
+	customRuleYaml, _ := yaml.JSONToYAML([]byte(customRuleSchemaYamlFileContent))
+
+	errorsResult, _ := jsonSchemaValidator.ValidateYamlSchema(string(customRuleYaml), string(resourceYaml))
+
+	assert.GreaterOrEqual(t, len(errorsResult), 1)
```

Comment:
can we assert that errorsResult[0].Error is equal to the error string? up to you